src/routes/docs/tutorials/android/step-4/+page.markdoc

```diff
@@ -0,0 +1,509 @@
+    android:inputType="textEmailAddress"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toTopOf="parent" />
+```
+
+- Password field
+
+```xml
+<EditText
+    android:id="@+id/password_edit_text"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:hint="Password"
+    android:inputType="textPassword"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@+id/email_edit_text" />
+```
+
+- Login button to submit the entered email and password
+
+```xml
+<Button
+    android:id="@+id/login_button"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:text="Login"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@+id/password_edit_text" />
+```
+
+- Register button to launch **RegisterActivity**
+
+```xml
+<Button
+    android:id="@+id/register_button"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:text="REGISTER?"
+    app:layout_constraintEnd_toEndOf="parent"
+    app:layout_constraintStart_toStartOf="parent"
+    app:layout_constraintTop_toBottomOf="@+id/login_button" />
+```
+
+## **LoginActivity.java**
+
+Now let us write logic of button clicks. There are a couple of ways to implement `OnClickListener` on views. Here we are going to use `View.OnClickListener` interface.
+
+Change the definition of `LoginActivity` to implement `View.OnClickListener` interface.
+
+```java
+public class LoginActivity extends AppCompatActivity implements View.OnClickListener
+```
+
+Declare UI views within this class
+
+```java
+EditText emailEditText, passwordEditText;
+Button loginButton, registerButton;
+```
+
+Initialize these variables in `onCreate` method.
+
+```java
+emailEditText = findViewById(R.id.email_edit_text);
+passwordEditText = findViewById(R.id.password_edit_text);
+loginButton = findViewById(R.id.login_button);
+registerButton = findViewById(R.id.register_button);
+```
+
+Set `OnClickListener`s on both the buttons inside `onCreate` method.
+
+```java
+loginButton.setOnClickListener(this);
+registerButton.setOnClickListener(this);
+```
+
+Now override the `onClick` method of `LoginActivity` to finally write the click logic.
+
+```java
+@Override
+public void onClick(View view) {
+
+    /* Extract user input */
+    String email = emailEditText.getText().toString();
+    String password = passwordEditText.getText().toString();
+
+    switch ((view.getId())) {
+        case R.id.login_button:
+            /* Do something */
+            break;
+
+        case R.id.register_button:
+            /* Do something */
+            break;
+    }
+}
+```
+
+We can simply use our Appwrite method to login the user, if **login_button** is pressed -
+
+```java
```

Comment:
The way you present this code is really hard to follow, I can't tell where this code goes, what imports I need where, what the file structure of the project is.

Can we simplify this so that we can have a more minimalistic authentication component in 1-2 smaller files?

Reply:
I have simplified authentication to be in just one activity now.